File: llm_generated_code/ChatGPT/Python/Level2/Q4.py
def max_profit(prices):
    if not prices:
        return 0

    min_price = float('inf')
    max_profit = 0

    for price in prices:
        # Update min_price if we find a new lower price
        if price < min_price:
            min_price = price
        # Calculate profit if we sell at the current price
        elif price - min_price > max_profit:
            max_profit = price - min_price

    return max_profit

# Example usage
prices = [7, 1, 5, 3, 6, 4]
profit = max_profit(prices)
print("Maximum profit Rohan can achieve:", profit)

File: llm_generated_code/ChatGPT/CPP/Level2/Q4.cpp
#include <iostream>
#include <vector>
#include <limits.h>
using namespace std;

int maxProfit(vector<int>& prices) {
    int minPrice = INT_MAX;
    int maxProfit = 0;

    for (int price : prices) {
        if (price < minPrice) {
            minPrice = price;  // Update minimum price if a lower price is found
        }
        else if (price - minPrice > maxProfit) {
            maxProfit = price - minPrice;  // Update max profit if current profit is higher
        }
    }

    return maxProfit;
}

int main() {
    vector<int> prices = {7, 1, 5, 3, 6, 4};  // Example price changes

    int profit = maxProfit(prices);

    cout << "Maximum profit Rohan can achieve: " << profit << endl;

    return 0;
}

